src/EditorFeatures/Core/Implementation/EditAndContinue/ActiveStatementTrackingService.cs

```diff
@@ -188,20 +190,47 @@ private async Task TrackActiveSpansAsync()
             {
                 try
                 {
-                    var baseActiveStatements = await _editSession.BaseActiveStatements.GetValueAsync(_editSession.CancellationToken).ConfigureAwait(false);
+                    var cancellationToken = _editSession.CancellationToken;
+                    var baseActiveStatements = await _editSession.BaseActiveStatements.GetValueAsync(cancellationToken).ConfigureAwait(false);
+                    var lastCommittedSolution = _editSession.DebuggingSession.LastCommittedSolution;
+                    var currentSolution = _editSession.DebuggingSession.Workspace.CurrentSolution;
+                    var activeSpansToTrack = ArrayBuilder<(Document, Document, ITextSnapshot, ImmutableArray<ActiveStatement>)>.GetInstance();
+
+                    foreach (var (documentId, documentActiveStatements) in baseActiveStatements.DocumentMap)
+                    {
+                        var document = currentSolution.GetDocument(documentId);
+                        if (document == null)
+                        {
+                            // Document has been deleted.
+                            continue;
+                        }
+
+                        var (baseDocument, _) = await lastCommittedSolution.GetDocumentAndStateAsync(documentId, cancellationToken).ConfigureAwait(false);
+                        if (baseDocument == null)
+                        {
+                            // Document has been added, is out-of-sync or a design-time-only document.
+                            continue;
+                        }
+
+                        if (!TryGetSnapshot(document, out var snapshot))
+                        {
+                            // Document is not open in an editor or a corresponding snapshot doesn't exist anymore.
+                            continue;
+                        }
+
+                        activeSpansToTrack.Add((baseDocument, document, snapshot, documentActiveStatements));
```

Comment:
Comment might be needed: why do we collect them all and then do them under the lock once we're done? The old code didn't and not obvious why the change.